test/tests/crypto/Test_RangeProofs.cpp

```diff
@@ -1,9 +1,43 @@
+#include <catch.hpp>
 
+#include <mw/crypto/Crypto.h>
+#include <mw/crypto/Random.h>
 
-/* TEST:
- *      1. Create a RangeProof via Crypto::GenerateRangeProof. Use the same value for privateNonce and rewindNonce.
- *      2. Try rewinding it via Crypto::RewindRangeProof using the *wrong* rewindNonce. Make sure it returns null.
- *      3. Try rewinding it via Crypto::RewindRangeProof using the *correct* rewindNonce (the one you passed to GenerateRangeProof). Make sure it returns a valid RewoundProof.
- *      4. Make sure amount, blindingFactor (aka 'key'), and ProofMessage match the values passed to GenerateRangeProof
- *      5. Make sure VerifyRangeProofs returns true. Use an empty vector for the third tuple value.
- */
\ No newline at end of file
+TEST_CASE("Range Proofs")
+{
+    const uint64_t value = 123;
+    BlindingFactor blind = Random::CSPRNG<32>();
+    Commitment commit = Crypto::CommitBlinded(value, blind);
+    SecretKey nonce = Random::CSPRNG<32>();
+    SecretKey nonce2;
+    do {
```

Comment:
What's with this do while? You should just be able to set nonce = nonce.

Reply:
O, I see you're using nonce to make sure you can't rewind with it. You still shouldn't need the do...while. The random number generator should be trusted, since we'll have tests for it elsewhere.